### PR TITLE
Working on conditional and filter parsing

### PR DESCRIPTION
**Pull request by jherico**

### CODE REVIEW

File: interface/resources/controllers/hydra.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
     "name": "Hydra to Standard",
     "channels": [
-        { "from": "Hydra.LY", "to": "Standard.LY" }, 
+        { "from": "Hydra.LY", "filters": "invert", "to": "Standard.LY" },
```
Comment (jherico) [inline interface/resources/controllers/hydra.json]: For hydra +Y is pressing forward on the stick, for SDL +Y is pulling back.  So we need to invert one of them.  

On the one hand my intuition says forward is +Y, but on the other hand, SDL2 is probably the most common API for interacting with joysticks (outside of DirectX), so I'm going with SDL's interpretation.

File: interface/resources/controllers/standard.json [hunk 2]
```diff
@@ -5,38 +5,25 @@
...
-        },
-        { 
-            "from": "Standard.DL", 
-            "to": "Actions.LATERAL_LEFT",
-            "filters": [ { "type": "scale", "scale": 0.5 } ]
-        },
-        { "from": "Standard.Y", "to": "Actions.VERTICAL_UP" },
-        { "from": "Standard.X", "to": "Actions.VERTICAL_DOWN" },
+
+        { "from": [ "Standard.DU", "Standard.DU", "Standard.DU", "Standard.DD" ], "to": "Standard.LeftPrimaryThumb" },
```
Comment (jherico) [inline interface/resources/controllers/standard.json]: This doesn't seem to be working quite right.  I can't see the indicator activate in the test dialog (Ctrl-Shift-X)

File: interface/resources/controllers/standard.json [hunk 3]
```diff
@@ -5,38 +5,25 @@
...
+        { "from": "Standard.Back", "to": "Standard.LeftSecondaryThumb" },
+
+        { "from": [ "Standard.A", "Standard.B", "Standard.X", "Standard.Y" ], "to": "Standard.RightPrimaryThumb" },
+        { "from": "Standard.Start", "to": "Standard.RightSecondaryThumb" },
+
         { 
             "from": "Standard.RT", 
             "to": "Actions.BOOM_IN", 
             "filters": [ { "type": "scale", "scale": 0.1 } ]
```
Comment (jherico) [inline interface/resources/controllers/standard.json]: Still way too fast.

File: interface/resources/qml/controller/AnalogButton.qml [hunk 4]
```diff
@@ -13,7 +13,7 @@ Item {
     property color color: 'black'
 
     function update() {
-        value = Controller.getValue(controlId);
+        value = controlId ? Controller.getValue(controlId) : 0;
```
Comment (jherico) [inline interface/resources/qml/controller/AnalogButton.qml]: prevent log spam on the test dialog when a device isn't connected / active

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1120554c5da4427564cc755859e8449cb76090ce available at: https://gist.github.com/7052d9e217b789c05560<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6141.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6141.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 97dd6786f2c8d0cb927e4312262f4bf9fab3c4c3 available at: https://gist.github.com/4b62d75edacbbd2c3b9d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6141.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6141.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b9b03bd842e03a60ba447c4039c2583b08bc1e33 available at: https://gist.github.com/4b62d75edacbbd2c3b9d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6141.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6141.dmg

**Pull request merged**